### PR TITLE
Bugfix FXIOS-5439 [v111] Focus jumps one tab ahead too many when closing a tab

### PR DESCRIPTION
**Pull request by thatswinnie**
[FXIOS-5439 ](https://mozilla-hub.atlassian.net/browse/FXIOS-5439)
#12690

### CODE REVIEW

**Comment by isabelrios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Could not get the BITRISE_XCRESULT_PATH to generate code coverage
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Edited 6 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 49b63020772d9cf41d7c8e0a24b501557350d76f
</p>

File: Client/Frontend/Browser/GridTabViewController.swift [hunk 1]
```diff
@@ -36,7 +36,7 @@ class GridTabViewController: UIViewController, TabTrayViewDelegate, Themeable {
     let profile: Profile
     weak var delegate: TabTrayDelegate?
     var tabDisplayManager: TabDisplayManager!
-    var tabCellIdentifer: TabDisplayer.TabCellIdentifer = TabCell.cellIdentifier
```
Comment (nbhasin2) [inline Client/Frontend/Browser/GridTabViewController.swift:39]: good catch

File: Client/Frontend/Browser/Tab Management/TabManager.swift [hunk 2]
```diff
@@ -601,8 +595,11 @@ class TabManager: NSObject, FeatureFlaggable, TabManagerProtocol {
     func removeTab(_ tab: Tab, completion: (() -> Void)? = nil) {
         guard let index = tabs.firstIndex(where: { $0 === tab }) else { return }
         DispatchQueue.main.async { [unowned self] in
+            // gather the index of the deleted tab within the viable tabs array
+            // so we can select the correct next tab after deletion
+            let viableTabsIndex = deletedIndexForViableTabs(tab)
             self.removeTab(tab, flushToDisk: true)
-            self.updateIndexAfterRemovalOf(tab, deletedIndex: index)
+            self.updateIndexAfterRemovalOf(tab, deletedIndex: index, viableTabsIndex: viableTabsIndex)
```
Comment (nbhasin2) [inline Client/Frontend/Browser/Tab Management/TabManager.swift:602]: Q. Will this show inactive tabs as part of top tab tray?

Comment (nbhasin2) [reply Client/Frontend/Browser/Tab Management/TabManager.swift:602]: Remember the discussion about inactive tabs on top tab tray in iPad 🤔

Comment (thatswinnie) [reply Client/Frontend/Browser/Tab Management/TabManager.swift:602]: With these changes inactive tabs are not shown anymore in top tabs.

**Review by nbhasin2 — APPROVED** [review]
overall lgtm

**Pull request merged**